modules/basic-skills/src/backend/slot.ts

```diff
@@ -0,0 +1,153 @@
+import * as sdk from 'botpress/sdk'
+import { Transition } from './typings'
+
+const generateFlow = async (data: any, metadata: sdk.FlowGeneratorMetadata): Promise<sdk.FlowGenerationResult> => {
+  return {
+    transitions: createTransitions(),
+    flow: {
+      startNode: 'check-if-extracted',
+      nodes: createNodes(data),
+      catchAll: {
+        next: []
+      }
+    }
+  }
+}
+
+const createTransitions = (): Transition[] => {
+  return [
+    { caption: 'On extracted', condition: 'temp.extracted == "true"', node: '' },
+    { caption: 'On not found', condition: 'temp.notExtracted == "true"', node: '' },
+    { caption: 'On already extracted', condition: 'temp.alreadyExtracted == "true"', node: '' }
+  ]
+}
+
+const createNodes = data => {
+  const slotExtractOnReceive = [
+    {
+      type: sdk.NodeActionType.RunAction,
+      name: `basic-skills/slot_fill {"slotName":"${data.slotName}","entity":"${data.entity}"}`
+    }
```

Comment:
Is there a reason for creating manually the skill parameters ? You could move the arguments in an object and let the backend prepare it itself:


```js
{
  type: sdk.NodeActionType.RunAction,
  name: 'basic-skills/slot_fill',
  args: {
    slotName: data.slotName,
    entity: data.entity
  }
}
```

![image](https://user-images.githubusercontent.com/42552874/55928512-ea1ebf00-5be6-11e9-8350-02e19a110b84.png)

Reply:
wow this is much better, thanks!